src/common/buf.rs

```diff
@@ -0,0 +1,75 @@
+use std::collections::VecDeque;
```

Comment:
can we move these to an external crate? I'd like to use this in tonic.

Reply:
I've been wanting something like this externally too. @carllerche suggested maybe in something like `bytes-util` or `tokio-codec` or somewhere. 🤷‍♂